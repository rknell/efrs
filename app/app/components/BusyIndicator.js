import React, { Component } from 'react'
import { ActivityIndicator, View } from 'react-native'
import { observer } from 'mobx-react/native'
import GlobalStyle from '../services/GlobalStyle'

@observer
class BusyIndicator extends Component {
  render () {
    if (this.props.isBusy) {
      return (
        <View style={{
          position: 'absolute',
          top: 0,
          right: 0,
          left: 0,
          bottom: 0,
          alignItems: 'center',
          justifyContent: 'center',
          backgroundColor: 'rgba(0,0,0,0.5)'
        }}>
          <View style={{
            width: 100,
            height: 100,
            backgroundColor: GlobalStyle.style.primaryColour,
            borderRadius: 10,
            alignItems: 'center',
            justifyContent: 'center'
          }}>
            <ActivityIndicator size='large' color={GlobalStyle.style.primaryColourText} />
          </View>
        </View>
      )
    } else {
      return null
    }
  }
}

export default BusyIndicator
