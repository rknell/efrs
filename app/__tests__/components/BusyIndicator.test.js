import React from 'react'
import BusyIndicator from '../../app/components/BusyIndicator'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

it('should match snapshot', () => {
  const wrapper = shallow(<BusyIndicator />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should show a busy indicator', () => {
  const wrapper = shallow(<BusyIndicator isBusy />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
