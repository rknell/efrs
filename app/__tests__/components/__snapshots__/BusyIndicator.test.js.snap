// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `""`;

exports[`should show a busy indicator 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "borderRadius": 10,
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="rgb(255, 255, 255)"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
